app/src/main/cpp/skyline/services/socket/sfdnsres/IResolver.cpp

```diff
@@ -2,7 +2,178 @@
 
 #include "IResolver.h"
+#include <arpa/inet.h>
+#include <common/settings.h>
 
 namespace skyline::service::socket {
+    static NetDbError AddrInfoErrorToNetDbError(i32 result) {
+        switch (result) {
+            case 0:
+                return NetDbError::Success;
+            case EAI_AGAIN:
+                return NetDbError::TryAgain;
+            case EAI_NODATA:
+                return NetDbError::NoData;
+            default:
+                return NetDbError::HostNotFound;
+        }
+    }
+
     IResolver::IResolver(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager) {}
+
+    Result IResolver::GetAddrInfoRequest(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto [dataSize, resultCode]{GetAddrInfoRequestImpl(request)};
+        response.Push<i32>(resultCode);  // errno
+        response.Push(AddrInfoErrorToNetDbError(resultCode));  // NetDBErrorCode
+        response.Push<u32>(dataSize);
+        return {};
+    }
+
+    Result IResolver::GetHostByNameRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return {};
+    }
+
+    Result IResolver::GetAddrInfoRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto [dataSize, resultCode]{GetAddrInfoRequestImpl(request)};
+        response.Push<i32>(resultCode);  // errno
+        response.Push(AddrInfoErrorToNetDbError(resultCode));  // NetDBErrorCode
+        response.Push<u32>(dataSize);
+        response.Push<u32>(0);
+        return {};
+    }
+
+    Result IResolver::GetNameInfoRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return {};
+    }
+
+    std::pair<u32, i32> IResolver::GetAddrInfoRequestImpl(ipc::IpcRequest &request) {
+        auto hostname{request.inputBuf.at(0).as_string(true)};
+        auto service{request.inputBuf.at(1).as_string(true)};
+
+        if (!(*state.settings->isInternetEnabled)) {
+            Logger::Info("Internet access disabled, DNS Blocked: {}", hostname);
+            return {0, -1};
+        }
+
+        addrinfo *result;
+        i32 resultCode = getaddrinfo(hostname.data(), service.data(), nullptr, &result);
+
+        u32 dataSize{0};
+        if (resultCode == 0 && result != nullptr) {
+            const std::vector<u8> data = SerializeAddrInfo(result, resultCode, hostname);
+            dataSize = static_cast<u32>(data.size());
+            request.outputBuf.at(0).copy_from(data);
+            freeaddrinfo(result);
+        }
+        return {dataSize, resultCode};
+    }
+
+    // https://github.com/yuzu-emu/yuzu/blob/ce8f4da63834be0179d98a7720dee47d65f3ec06/src/core/hle/service/sockets/sfdnsres.cpp#L76
+    std::vector<u8> IResolver::SerializeAddrInfo(const addrinfo* addrinfo, i32 result_code,
+                                      std::string_view host) {
```

Comment:
You can reformat code you take from yuzu IMO, put all function arguments on the same line.

Reply:
Yeah, this needs a bunch of changed I just haven't reviewed yet